File: node/test/smoke.test.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { machineCode, sign, LicenseClient } from '../src/index.js';

test('machineCode stable & long enough', () => {
  assert.equal(machineCode(), machineCode());
  assert.ok(machineCode().length >= 8);
});

test('sign deterministic', () => {
  const params = { productUniqueCode: 'PRO-2026-001', machineCode: 'M123', edition: 'PRO', expiryDays: 0, clientOrderId: 'x', licenseCode: '', timestamp: 1000 };
  assert.equal(sign('secret', params), sign('secret', params));
});

test('purchaseUrl carries params', () => {
  const c = new LicenseClient({ productUniqueCode: 'PRO-2026-001' });
  const url = c.purchaseUrl('MABC', { base: 'https://www.powersoftware.app' });
  assert.match(url, /productUniqueCode=PRO-2026-001/);
  assert.match(url, /machineCode=MABC/);
  assert.doesNotMatch(url, /productId=/);
});

test('checkUpdate posts updateCheck and parses content', async () => {
  let captured;
  const fetchImpl = async (url, init) => {
    captured = { url, body: JSON.parse(init.body) };
    return new Response(JSON.stringify({ success: true, content: { hasUpdate: true, latestVersion: '1.3.0' } }), { status: 200 });
  };
  const c = new LicenseClient({ productUniqueCode: 'PRO-2026-001', fetchImpl });
  const result = await c.checkUpdate('1.2.0');
  assert.match(captured.url, /\/product\/updateCheck$/);
  assert.equal(captured.body.productUniqueCode, 'PRO-2026-001');
  assert.equal(captured.body.currentVersion, '1.2.0');
  assert.deepEqual(result, { hasUpdate: true, latestVersion: '1.3.0' });
});

test('checkUpdate requires productUniqueCode', () => {
  const c = new LicenseClient({});
  assert.throws(() => c.checkUpdate('1.2.0'), /productUniqueCode required/);
});

File: java/src/test/java/com/powersoftware/sdk/SmokeTest.java
package com.powersoftware.sdk;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 冒烟测试（无 JUnit 依赖，直接 main 运行）：
 * java -cp target/classes com.powersoftware.sdk.SmokeTest
 */
public class SmokeTest {

    public static void main(String[] args) {
        String mc1 = LicenseClient.machineCode();
        String mc2 = LicenseClient.machineCode();
        check(mc1.equals(mc2), "machineCode stable");
        check(mc1.length() >= 8, "machineCode length");

        Map<String, Object> params = new LinkedHashMap<>();
        params.put("productId", 1);
        params.put("machineCode", "M123");
        params.put("edition", "PRO");
        params.put("expiryDays", 0);
        params.put("clientOrderId", "x");
        params.put("licenseCode", "");
        params.put("timestamp", 1000L);
        check(LicenseClient.sign("secret", params).equals(LicenseClient.sign("secret", params)), "sign deterministic");

        String url = new LicenseClient(88, "secret").purchaseUrl("MABC");
        check(url.contains("productId=88") && url.contains("machineCode=MABC"), "purchaseUrl params");
        String url2 = new LicenseClient(null, "secret").purchaseUrl("MABC", "https://www.powersoftware.cn", "PRO-2026-001");
        check(url2.contains("productUniqueCode=PRO-2026-001") && url2.contains("machineCode=MABC") && !url2.contains("productId="), "purchaseUrl productUniqueCode");
        System.out.println("SmokeTest OK");
    }

    private static void check(boolean cond, String name) {
        if (!cond) {
            throw new AssertionError("FAILED: " + name);
        }
        System.out.println("ok: " + name);
    }
}
